xsl/tokenize.xsl

```diff
@@ -116,157 +117,115 @@
-                    <xsl:apply-templates select="$tokenizedDoc" mode="enumerate"/>
-                </xsl:result-document>
-         
+            <xsl:variable name="cleaned">
+                <xsl:apply-templates select="$excluded" mode="clean"/>
+            </xsl:variable>
+            
+            <!--Next add weighting information to the cleaned document-->
+            <xsl:variable name="weighted">
+                <xsl:apply-templates select="$cleaned" mode="weigh"/>
+            </xsl:variable>
+            
+            <!--Next add context information to the weighted document-->
+            <xsl:variable name="contextualized">
+                <xsl:apply-templates select="$weighted" mode="contextualize"/>
+            </xsl:variable>
             
-                <!--If we're running in verbose mode, then output all of the interstitial
+            <!--Next tokenize and stem the contextualized document-->
+            <xsl:variable name="tokenizedDoc">
+                <xsl:apply-templates select="$contextualized" mode="tokenize">
+                    <xsl:with-param name="currDocUri" select="$uri" tunnel="yes"/>
+                </xsl:apply-templates>
+            </xsl:variable>
+            
+            <!--And finally pass the tokenized document through the enumeration templates-->
+            <xsl:apply-templates select="$tokenizedDoc" mode="enumerate"/>
+            
+            <!--If we're running in verbose mode, then output all of the interstitial
                     documents for easier debugging.-->
-                <xsl:if test="$verbose">
-                    <xsl:message>Creating <xsl:value-of select="$cleanedOutDoc"/></xsl:message>
-                    <xsl:result-document href="{$cleanedOutDoc}">
-                        <xsl:copy-of select="$cleaned"/>
-                    </xsl:result-document>
-                    <xsl:message>Creating <xsl:value-of select="$contextualizedOutDoc"/></xsl:message>
-                    <xsl:result-document href="{$contextualizedOutDoc}">
-                        <xsl:copy-of select="$contextualized"/>
+            <xsl:if test="$verbose">
+                <!--Stash all of the documents we want to output into a map so we can simply
+                    iterate through them-->
+                <xsl:variable name="outputMap" select="map{
+                    'cleaned': $cleaned,
+                    'contextualized': $contextualized,
+                    'weighted': $weighted,
+                    'excluded': $excluded
+                    }"/>
+                <!--Iterate through the keys, which are the filenames-->
+                <xsl:for-each select="map:keys($outputMap)">
+                    <xsl:result-document href="{replace(current-output-uri(),'_tokenized',('_' || .))}">
+                        <xsl:message>Creating <xsl:value-of select="current-output-uri()"/></xsl:message>
+                        <xsl:copy-of select="$outputMap(.)"/>
```

Comment:
Shouldn't this be using map:get()?

Reply:
Either works—I tend to do the shorter version just for brevity, but I'll switch it to `map:get()` to be explicit.

Reply:
That was a learning process for me. I didn't realize it was OK to use function coercion to do that.  :-)